lib/app.js

```diff
@@ -642,15 +649,23 @@ module.exports = {
                                 'permission to use this Firebase');
           process.exit(1);
         }
+
         for (var i in defaultSettings) {
-          if (defaultSettings.hasOwnProperty(i) &&
+          if (i === 'public') {
+            if (settings.hasOwnProperty('public')) {
+              settings.public = path.resolve(defaultSettings.public, settings.public);
```

Comment:
The use of `defaultSettings.public` here is confusing - to make it more readable we should rename `curDir` to something more descriptive like `projectRoot` of `projectRootDir` and then we could use that directly here